src/Lookup/AdvancedFind.ts

```diff
@@ -11,59 +11,93 @@ import {
   RawNoErrorScalarValue
 } from '../interpreter/InterpreterValue'
 import {SimpleRangeValue} from '../SimpleRangeValue'
-import {SearchOptions} from './SearchStrategy'
+import {AdvancedFindOptions, SearchOptions} from './SearchStrategy'
 import {forceNormalizeString} from '../interpreter/ArithmeticHelper'
-import {findLastOccurrenceInOrderedRange} from '../interpreter/binarySearch'
+import {compare, findLastOccurrenceInOrderedRange} from '../interpreter/binarySearch'
+
+const NOT_FOUND = -1
 
 export abstract class AdvancedFind {
   protected constructor(
     protected dependencyGraph: DependencyGraph
   ) {
   }
 
-  public advancedFind(keyMatcher: (arg: RawInterpreterValue) => boolean, rangeValue: SimpleRangeValue): number {
-    let values: InternalScalarValue[]
+  public advancedFind(keyMatcher: (arg: RawInterpreterValue) => boolean, rangeValue: SimpleRangeValue, { returnOccurence }: AdvancedFindOptions = { returnOccurence: 'first' }): number {
```

Comment:
a typo :) 
```suggestion
  public advancedFind(keyMatcher: (arg: RawInterpreterValue) => boolean, rangeValue: SimpleRangeValue, { returnOccurence }: AdvancedFindOptions = { returnOccurrence: 'first' }): number {
```

Reply:
Thanks @budnix, great catch. I changed every occurrence of "occurence" (single r).